cmsroles/admin.py

```diff
@@ -70,6 +70,11 @@ def has_change_permission(self, request, obj=None):
         user = request.user
         return is_site_admin(user) and len(get_administered_sites(user)) > 0
 
+    def change_view(self, request, object_id=None, *args, **kwargs):
+        if object_id:
```

Comment:
The normal workflow should rely on https://docs.djangoproject.com/en/1.8/ref/contrib/admin/#django.contrib.admin.ModelAdmin.has_change_permission and then it would be obvious. The problem is that UserSetup is a hack, not a real model, so it will fail before reaching has_change_permission because Django tries to retrieve the object from the database.
I will add a comment to explain. Thanks.

We try to keep the naming consistent with Django:
https://docs.djangoproject.com/en/1.8/ref/contrib/admin/#django.contrib.admin.ModelAdmin.change_view